stan/math/prim/prob/multi_student_t_cholesky_lpdf.hpp

```diff
@@ -42,112 +49,324 @@ namespace math {
+                     size_mu);
   }
 
-  int num_dims = y_vec[0].size();
-
-  check_size_match(function, "Size of random variable", mu_vec[0].size(),
+  check_size_match(function, "Size of random variable", size_mu,
                    "rows of scale parameter", L.rows());
-  check_size_match(function, "Size of random variable", num_dims,
-                   "size of location parameter", mu_vec[0].size());
-  check_size_match(function, "Size of random variable", num_dims,
+  check_size_match(function, "Size of random variable", size_y,
+                   "size of location parameter", size_mu);
+  check_size_match(function, "Size of random variable", size_y,
                    "rows of scale parameter", L.rows());
-  check_size_match(function, "Size of random variable", num_dims,
+  check_size_match(function, "Size of random variable", size_y,
                    "columns of scale parameter", L.cols());
 
-  if (unlikely(num_dims == 0)) {
-    return lp_type(0);
-  }
-
   for (size_t i = 0; i < size_vec; i++) {
     check_finite(function, "Location parameter", mu_vec[i]);
     check_not_nan(function, "Random variable", y_vec[i]);
   }
 
-  check_cholesky_factor(function, "scale parameter", L_ref);
+  if (unlikely(num_dims == 0)) {
+    return T_return(0);
+  }
 
-  lp_type lp(0);
+  auto ops_partials = make_partials_propagator(y_ref, nu_ref, mu_ref, L_ref);
 
-  if (include_summand<propto, T_dof>::value) {
-    lp += lgamma(0.5 * (nu + num_dims)) * size_vec;
-    lp += -lgamma(0.5 * nu) * size_vec;
-    lp += -(0.5 * num_dims) * log(nu) * size_vec;
+  T_partials_return lp(0);
+  if (include_summand<propto>::value) {
+    lp += -0.5 * num_dims * LOG_PI * size_vec;
   }
 
-  if (include_summand<propto>::value) {
-    lp += -(0.5 * num_dims) * LOG_PI * size_vec;
+  if (include_summand<propto, T_y, T_dof, T_loc, T_covar_elem>::value) {
+    T_partials_return nu_val = value_of(nu_ref);
+    T_partials_return inv_nu = inv(nu_val);
+    T_partials_return nu_plus_dims = nu_val + num_dims;
+    const auto& half_nu
+        = to_ref_if<include_summand<propto, T_dof>::value>(0.5 * nu_val);
+
+    Eigen::Matrix<T_partials_return, Eigen::Dynamic, Eigen::Dynamic>
+        y_val_minus_mu_val(size_y, size_vec);
+
+    for (size_t i = 0; i < size_vec; i++) {
+      vector_partials_t y_val = as_value_column_vector_or_scalar(y_vec[i]);
+      vector_partials_t mu_val = as_value_column_vector_or_scalar(mu_vec[i]);
+      y_val_minus_mu_val.col(i) = eval(y_val - mu_val);
+    }
+
+    matrix_partials_t L_val = value_of(L_ref);
+    matrix_partials_t half
+        = mdivide_left_tri<Eigen::Lower>(L_val, y_val_minus_mu_val).transpose();
+    matrix_partials_t scaled_diff
+        = mdivide_right_tri<Eigen::Lower>(half, L_val).transpose();
+
+    const auto& digamma_vals = to_ref_if<!is_constant<T_dof>::value>(
+        digamma(half_nu + 0.5 * num_dims) - digamma(half_nu));
+    matrix_partials_t L_deriv(num_dims, num_dims);
+    L_deriv.setZero();
```

Comment:
@SteveBronder I couldn't find a way to conditionally create this partial matrix. I wanted to do something like
```
const auto& L_deriv
        = to_ref_if<include_summand<propto, T_covar_elem>::value>(Eigen::MatrixXd::Zero(num_dims, num_dims));
```
Since this matrix only needs to get created when that condition passes. However, I can't just wrap it in a conditional because Eigen complains that the matrix isn't available further down the code in the loop.